File: data/EXP01/golang_no_system_var/2491/2491.output_3.go
func dividePlayers(skill []int) int {
    totalSkills := 0
    for _, s := range skill {
        totalSkills += s
    }
    if totalSkills%2 != 0 {
        return -1
    }
    
    sort.Ints(skill)
    
    n := len(skill)
    sum := 0
    for i := 0; i < n/2; i++ {
        sum += skill[i] * skill[n-i-1]
    }
    
    return sum
}

File: data/EXP01/python/2491/2491.output_3.py
# @lc app=leetcode id=2491 lang=python3
class Solution:
    def dividePlayers(self, skill):
        n = len(skill)
        total_sum = sum(skill)
        if total_sum % (n // 2) != 0:
            return -1
        
        team_sum = total_sum // (n // 2)
        skill.sort(reverse=True)
        
        chemistries = []
        i, j = 0, n-1
        while i < j:
            if skill[i] + skill[j] != team_sum:
                return -1
            chemistries.append(skill[i] * skill[j])
            i += 1
            j -= 1
        
        return sum(chemistries)